<launch>
  <arg name="cam_name" default="csi_cam_0" />
  <arg name="cam_device" default="/dev/video0" />
  <arg name="gst_pipeline" value="nvarguscamerasrc ! video/x-raw(memory:NVMM), width=1920, height=1080, framerate=30/1, format=NV12 !
                                  nvvidconv flip-method=2 ! video/x-raw, format=BGRx ! videoconvert ! video/x-raw, format=BGR ! appsink" />

  <node pkg="gscam" type="gscam" name="$(arg cam_name)">
    <param name="camera_name" value="$(arg cam_name)" />
    <param name="gscam_config" value="$(arg gst_pipeline)" />
    <param name="camera_info_url" value="file://$(find fisheye_csi_cam)/config/camera_calibration.yaml" />
    <remap from="camera/image_raw" to="$(arg cam_name)/image_raw" />
  </node>

  <node pkg="image_view" type="image_view" name="raw_view" >
    <remap from="image" to="/$(arg cam_name)/image_raw" />
  </node>

  <node pkg="image_view" type="image_view" name="undistorted_view" >
    <remap from="image" to="/image_undistorted" />
  </node>

  <node pkg="fisheye_csi_cam" type="undistorted_node.py" name="undistort" output="screen" />

</launch>